wicket-core/src/main/java/org/apache/wicket/authentication/strategy/DefaultAuthenticationStrategy.java

```diff
@@ -57,25 +65,50 @@
 	 * 
 	 * @param cookieKey
 	 *            The name of the cookie
+	 *            
+	 * @deprecated supply a crypt instead TODO remove in Wicket 10
 	 */
+	@Deprecated
 	public DefaultAuthenticationStrategy(final String cookieKey)
 	{
-		this(cookieKey, defaultEncryptionKey(cookieKey));
+		this(cookieKey, defaultEncryptionKey());
 	}
 
-	private static String defaultEncryptionKey(String cookieKey)
+	private static String defaultEncryptionKey()
 	{
-		if (Application.exists())
-		{
-			return Application.get().getName();
-		}
-		return cookieKey;
+		return UUID.randomUUID().toString();
 	}
 
+	/**
+	 * @deprecated supply a crypt instead TODO remove in Wicket 10
+	 */
+	@Deprecated
 	public DefaultAuthenticationStrategy(final String cookieKey, final String encryptionKey)
+	{
+		this(cookieKey, defaultCrypt(encryptionKey));
+	}
+
+	private static ICrypt defaultCrypt(String encryptionKey)
+	{
+		byte[] salt = SunJceCrypt.randomSalt();
+
+		SunJceCrypt crypt = new SunJceCrypt(salt, 1000);
+		crypt.setKey(encryptionKey);
+		return crypt;
+	}
+
+	/**
+	 * Constructor
+	 * 
+	 * @param cookieKey
+	 *            The name of the cookie
+	 * @param crypt
+	 *            the crypt
+	 */
+	public DefaultAuthenticationStrategy(final String cookieKey, ICrypt crypt)

```

Comment:
I guess this is the stable constructor that should be used by the applications if they want ti be able to decrypt after restarts. Let's add a comment to make it clear/recommended.